kernel/common/enumerator.rb

```diff
@@ -148,7 +149,19 @@ def rewind
     end
 
     def size
-      @size.kind_of?(Proc) ? @size.call : @size
+      if @size.kind_of?(Proc)
+        @size.call
+      elsif @size.is_a? Numeric
+        @size
+      elsif @size.respond_to? :size
+        @size.size
+      else
```

Comment:
What is the reason for changing this method itself? As far as I'm aware `@size` is always either a Numeric or a Proc, thus it will never respond to `size`. Checking if it's a `Numeric` also isn't really needed since we can guarantee the types of `@size`.